core/lib/l1_contract_interface/src/i_executor/structures/commit_batch_info.rs

```diff
@@ -92,24 +147,95 @@ impl<'a> Tokenizable for CommitBatchInfo<'a> {
                 ),
                 // `eventsQueueStateHash`
                 Token::FixedBytes(
-                    self.0
+                    self.l1_batch_with_metadata
                         .metadata
                         .events_queue_commitment
                         .unwrap()
                         .as_bytes()
                         .to_vec(),
                 ),
                 // `systemLogs`
-                Token::Bytes(serialize_commitments(&self.0.header.system_logs)),
-                // `totalL2ToL1Pubdata`
+                Token::Bytes(serialize_commitments(
+                    &self.l1_batch_with_metadata.header.system_logs,
+                )),
+            ]
+        }
+    }
+}
+
+impl<'a> Tokenizable for CommitBatchInfo<'a> {
+    fn from_token(_token: Token) -> Result<Self, Web3ContractError>
+    where
+        Self: Sized,
+    {
+        // Currently there is no need to decode this struct.
+        // We still want to implement `Tokenizable` trait for it, so that *once* it's needed
+        // the implementation is provided here and not in some other inconsistent way.
+        Err(Web3ContractError::Api(Web3ApiError::Decoder(
+            "Not implemented".to_string(),
+        )))
+    }
+
+    fn into_token(self) -> Token {
+        let mut tokens = self.base_tokens();
+
+        if self
+            .l1_batch_with_metadata
+            .header
+            .protocol_version
+            .unwrap()
```

Comment:
You can't unwrap protocol_version, instead use unwrap_or_else with ProtocolVersionId::last_potentially_undefined